### PR TITLE
VQE supports initialization by computer

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
Addresses a major use case of #168 - VQE now allows the user to pass in a computer to initialize the state.

## User Notes
- [x] VQE now supports initializing the wavefunction with a `Computer`.

## Checklist
- [x] Added/updated tests of new features
- [x] Ready to go!

### CODE REVIEW

**Comment by JonathonMisiewicz** [comment]
Note that on my machine, the newly added test takes about 20 seconds to run.

File: tests/test_computer_init.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        fci_energy = -2.162897881184882
+
+        computer = Computer(nqubits)
+        coeff_vec = np.zeros(2**nqubits)
+        coeff_vec[int("00001111", 2)] = 1
+        coeff_vec[int("00110011", 2)] = 0.2
+        coeff_vec[int("00111100", 2)] = 0.1
+        coeff_vec[int("11001100", 2)] = 0.04
+        coeff_vec /= np.linalg.norm(coeff_vec)
+        computer.set_coeff_vec(coeff_vec)
```
Comment (imagoulas) [inline tests/test_computer_init.py:31]: As an additional test, would it be possible to check that the initial guess energy agrees with the expectation value of the Hamiltonian with respect to the state generated by the computer?

**Review by imagoulas — COMMENTED** [review]
I have a few suggestions. Have a look and let me know what you think.

File: src/qforte/abc/algorithm.py [hunk 2]
```diff
@@ -93,6 +93,28 @@ def __init__(
             self._refprep = build_refprep(self._ref)
             self._Uprep = reference
 
+        elif self._state_prep_type == "computer":
+            if not isinstance(reference, qf.Computer):
+                raise ValueError("unitary_circ reference must be a Circuit.")
+            if not fast:
+                raise ValueError(
+                    "User specified they want to simulate a quantum computer but not specify how to prepare the initial state. That's inconsistent."
```
[the remark this answers is not in the document]
Comment (imagoulas) [reply src/qforte/abc/algorithm.py:101]: Perhaps the word `fast` is not the best descriptor for what the code is doing. Of course, this is outside the scope of this PR.

I understand that you want to be precise, but what worries me is that this error message does not reference the `fast` keyword. The user might not be able to immediately understand what they did to cause this error. For example, in other places of the code, similar error messages read: "self._fast must be True for gradient measurement."

**Review by imagoulas — APPROVED** [review]
I have one minor suggestion for you to consider.

File: src/qforte/abc/algorithm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (imagoulas) [reply src/qforte/abc/algorithm.py]: The new error message is great! Feel free to merge.

**Pull request merged**